include/RAJA/pattern/forall.hpp

```diff
@@ -589,6 +589,136 @@ RAJA_INLINE camp::resources::EventProxy<Res> CallForallIcount::operator()(T cons
 
 }  // namespace detail
 
+//
+// Experimental support for dynamic policy selection
+//
+namespace expt
+{
+
+
+
+template<typename T, typename t_pol, typename SEGMENT, typename BODY>
+struct IsHostResource {
+  static const bool value = false;
```

Comment:
I'm not sure about the naming of IsHostResource, where is value used?

Reply:
How much value is this abstraction adding? It seems like its mainly here to abstract our r.get<T>().

Reply:
Its keyed off the resource type not the policy, doesn't it always use the resource type given?

Reply:
That is right, but the resource that gets passed may not be compatible with a policy within the list. For example a cuda resource and a loop_exec policy. Perhaps you have something in mind that I don't quite see, lets chat about it.